tools/kliopt/org/jetbrains/kliopt/KliOpt.kt

```diff
@@ -172,25 +187,30 @@ class ArgParser(optionsList: List<OptionDescriptor>, argsList: List<ArgDescripto
         val name = nullArgs.firstOrNull()
         name?. let {
             argDescriptors.getValue(name).type.check(arg, name)
-            argDescriptors.getValue(name).deprecatedWarning?.let { println ("Warning: $it") }
+            argDescriptors.getValue(name).deprecated?.let { println ("${it.helpMessage(argDescriptors.getValue(name))}") }
             processedValues.getValue(name).add(arg)
             return true
         }
         return false
     }
 
-    private fun saveAsOption(descriptor: OptionDescriptor, value: String, processedValues: Map<String, MutableList<String>>) {
+    private fun saveAsOption(descriptor: OptionDescriptor, value: String,
+                             processedValues: Map<String, MutableList<String>>, useShort: Boolean = false) {
```

Comment:
`useShort` parameter name confuses. It is rather `shortNameUsed` or something like this.